Doc/library/ftplib.rst

```diff
@@ -59,25 +59,25 @@ FTP objects
                source_address=None, *, encoding='utf-8')
 
    Return a new instance of the :class:`FTP` class.
-   When *host* is given, the method call :meth:`connect(host) <connect>`
-   is made by the constructor.
-   When *user* is given, additionally the method call
-   :meth:`login(user, passwd, acct) <connect>` is made.
 
    :param str host:
       The hostname to connect to.
+      If given, :code:`connect(host)` is implicitly called by the constructor.
+      See the corresponding parameter to :meth:`connect` for more details.
 
    :param str user:
       The username to log in with.
-      If empty string, ``"anonymous"`` is used.
+      If given, :code:`login(host, passwd, acct)` is implicitly called
+      by the constructor.
+      See the corresponding parameter to :meth:`login` for more details.
 
    :param str passwd:
       The password to use when logging in.
-      If not given, and if *passwd* is the empty string or ``"-"``,
-      a password will be automatically generated.
+      See the corresponding parameter to :meth:`login` for more details.
 
    :param str acct:
-      Account information; see the ACCT FTP command.
+      Account information to be used for the ``ACCT`` FTP command.
+      See the corresponding parameter to :meth:`login` for more details.
```

Comment:
I'm a bit conflicted about this.  One one hand it makes sense to remove/avoid duplication, but on the other I think we are doing a disservice to the user.  This sentence sends me to the `login()` docs, that in turn send me to another RFC.  Moreover, the `login()`'s description ("See [RFC 959](https://datatracker.ietf.org/doc/html/rfc959.html) for more details. Few systems implement this.") is shorter than this sentence.

For this particular case it might be ok to duplicate the description, and possibly for others too (assuming the description is not overly long.

Another option to solve both problems could be using [placeholders](https://docutils.sourceforge.io/docs/ref/rst/directives.html#replacement-text), even though I haven't really seen them using like this before.  The idea is that you could have the description of common params in a single place, and then use the same definition in multiple places afterwards:

```rst
.. |acct_param| replace:: Account information to be used for the ``ACCT``
   FTP command. See :rfc:`959` for more details. Few systems implement this.

...

:param str acct:
   |acct_param|  
```

Reply:
We've done similar things over in the sqlite3 docs[^1] to reduce duplication, but I agree this PR is not optimal :( Placeholders seems to be a good solution (TIL).


[^1]: for example with `Connection.execute` (brief doc) and `Cursor.execute` (detailed doc)

Reply:
Resolved by using substitutions. What do you think? For some reason, Sphinx won't accept the `:rfc:` directive, so I had to spell out the RFC reference. I guess that is ok, since we link to it in the introduction.

Reply:
@AA-Turner, any idea why the `:rfc:` role is rejected by the `replace` directive?